nutils/function.py

```diff
@@ -1127,11 +1126,10 @@ def __init__(self, arg: Array, space: str) -> None:
         super().__init__(arg.shape, arg.dtype, arg.spaces, arg.arguments)
 
     def lower(self, args: LowerArgs) -> evaluable.Array:
-        opposite_args = LowerArgs(
-            args.points_shape,
-            tuple(LowerArg(arg.space, arg.transforms_list[::-1], arg.index, arg.coordinates) if arg.space == self._space else arg for arg in args.args),
-        )
-        return self._arg.lower(opposite_args)
+        opp_space = '~' + self._space
+        if self._space in args.spaces and opp_space in args.spaces:
```

Comment:
Shouldn't this be 'or' ?

Reply:
`function.opposite(arg)` wraps `arg` in `_Opposite(arg, space)` for each space in `arg`, regardless if this space has an opposite. This was already the case before this PR. If we'd swap `space` for `~space` if `~space` doesn't exist, `arg` becomes unevaluable. So we need `~space` to exist. Requiring `space` is not necessary, but prevents `opposite(opposite(arg))` to work.

Reply:
clear